modules/simple_forms_api/app/services/simple_forms_api/s3/submission_archive_builder.rb

```diff
@@ -0,0 +1,163 @@
+          include_text_archive: true, # include the form data as a text file
+          metadata: {}, # pertinent metadata for original file upload/submission
+          parent_dir: 'vff-simple-forms' # S3 bucket base directory where files live
+        }
+      end
+
+      def process_submission_files
+        write_pdf
+        write_as_json_archive if include_json_archive
+        write_as_text_archive if include_text_archive
+        write_attachments unless attachments.empty?
+        write_manifest if include_manifest
+        write_metadata
+      end
+
+      def write_pdf
+        write_tempfile(submission_pdf_filename, File.read(generate_pdf_content))
+      end
+
+      # TODO: this will be pulled out to be more team agnostic
+      def generate_pdf_content
+        return file_path if file_path
+
+        form_number = SimpleFormsApi::V1::UploadsController::FORM_NUMBER_MAP[submission.form_type]
+        form = "SimpleFormsApi::#{form_number.titleize.gsub(' ', '')}".constantize.new(form_data_hash)
+        filler = SimpleFormsApi::PdfFiller.new(form_number:, form:)
+
+        @file_path = filler.generate(timestamp: submission.created_at)
+        @metadata = SimpleFormsApiSubmission::MetadataValidator.validate(
+          form.metadata,
+          zip_code_is_us_based: form.zip_code_is_us_based
+        )
+
+        form.handle_attachments(file_path) if %w[vba_40_0247 vba_20_10207 vba_40_10007].include? form_number
+
+        @attachments = form.get_attachments if form_number == 'vba_20_10207'
+        @file_path
+      end
+
+      def form_data_hash
+        @form_data_hash ||= JSON.parse(submission.form_data)
+      end
+
+      def submission_pdf_filename
+        @submission_pdf_filename ||= "form_#{form_data_hash['form_number']}.pdf"
+      end
+
+      def error_details(error)
+        "#{error.message}\n\n#{error.backtrace.join("\n")}"
+      end
+
+      def write_as_json_archive
+        write_tempfile('form_json_archive.json', JSON.pretty_generate(form_data_hash))
+      end
+
+      def write_as_text_archive
+        form_data_hash['claim_date'] ||= submission.created_at.iso8601
+        write_tempfile('form_text_archive.txt', form_data_hash.to_s)
+      end
+
+      def write_metadata
+        write_tempfile('metadata.json', metadata.to_json)
+      end
+
+      def write_attachments
+        log_info("Processing #{attachments.count} attachments")
+        attachments.each_with_index { |upload, i| process_attachment(i + 1, upload) }
+        write_attachment_failure_report if attachment_failures.present?
+      rescue => e
+        handle_upload_error(e)
+      end
+
+      def process_attachment(attachment_number, guid)
+        log_info("Processing attachment ##{attachment_number}: #{guid}")
+        attachment = PersistentAttachment.find_by(guid:).to_pdf
+        raise 'Local record not found' unless attachment
+
+        write_tempfile("attachment_#{attachment_number}.pdf", attachment)
```

Comment:
I wonder if it'll make a difference but if we have non-PDF attachments could we just copy the file rather than using `File.write`? I dunno if a `copy` method would be more appropriate. Doesn't need to be figure out right away, though, I think.

Reply:
Yeah, admittedly this logic is very pdf centric. I should definitely change that in the next iteration. I'll leave a TODO in the code. Great catch!